assembl/static2/js/app/components/common/wordCloud/wordCloud.jsx

```diff
@@ -0,0 +1,165 @@
+// @flow
+import React, { Component } from 'react';
+
+import ReactWordCloud from 'react-wordcloud';
+
+import type { Keyword, Keywords } from '../../../integration/semanticAnalysis/dataType';
+
+// Keys provided by Watson
+const WORD_COUNT_KEY = 'relevance';
+const WORD_KEY = 'text';
+
+export type BaseProps = {
+  /** Optional angle value in degrees */
+  keywordsAngle: number,
+  /** Optional color rgb(x, x, x) only */
+  keywordsColor: string,
+  /** Optional color when a word is active rgb(x, x, x) only */
+  keywordsColorActive: string,
+  /** Array of keywords */
+  keywords: Keywords,
+  /** Optional maximum number of keywords to show */
+  numberOfKeywordsToDisplay: number,
+  /** Optional callback function called when a word is hovered in */
+  onMouseOutWord: (word: Keyword) => void,
+  /** Optional callback function called when a word is hovered out */
+  onMouseOverWord: (word: Keyword) => void,
+  /** Optional callback function called when a word is clicked */
+  onWordClick: (word: Keyword) => void
+};
+
+export type Props = BaseProps & {
+  /** Optional height */
+  height: number,
+  /** Optional width */
+  width: number
+};
+
+export const defaultBaseProps = {
+  keywordsAngle: 0,
+  keywordsColor: 'rgb(0, 0, 0)',
+  keywordsColorActive: 'rgb(0, 0, 0)',
+  numberOfKeywordsToDisplay: 30,
+  onMouseOutWord: () => {},
+  onMouseOverWord: () => {},
+  onWordClick: () => {}
+};
+
+class WordCloud extends Component<Props> {
+  static defaultProps = {
+    ...defaultBaseProps,
+    height: 500,
+    width: 400
+  };
+
+  shouldComponentUpdate(nextProps: Props) {
+    const {
+      height,
+      keywordsAngle,
+      keywordsColor,
+      keywordsColorActive,
+      keywords,
+      numberOfKeywordsToDisplay,
+      onMouseOutWord,
+      onMouseOverWord,
+      onWordClick,
+      width
+    } = this.props;
+    if (
+      nextProps.height !== height ||
+      nextProps.keywordsAngle !== keywordsAngle ||
+      nextProps.keywordsColor !== keywordsColor ||
+      nextProps.keywordsColorActive !== keywordsColorActive ||
+      nextProps.keywords !== keywords ||
+      nextProps.numberOfKeywordsToDisplay !== numberOfKeywordsToDisplay ||
+      nextProps.onMouseOutWord !== onMouseOutWord ||
+      nextProps.onMouseOverWord !== onMouseOverWord ||
+      nextProps.onWordClick !== onWordClick ||
+      nextProps.width !== width
```

Comment:
Same as above, I might be missing something but I believe you can refactor this block